lib/mongodb.nit

```diff
@@ -0,0 +1,701 @@
+			return false;
+		}
+		return true;
+	`}
+
+	# Removes the first document that matches `selector`.
+	#
+	# Returns `false` if an error occured. See `Sys::last_mongoc_error`.
+	#
+	# ~~~
+	# var client = new MongoClient("mongodb://localhost:27017/")
+	# var col = client.database("test").collection("test")
+	# var sel = new JsonObject
+	# sel["foo"] = 10
+	# assert col.remove(sel)
+	# client.close
+	# ~~~
+	fun remove(selector: JsonObject): Bool
+		import JsonObject.to_bson, set_mongoc_error `{
+		bson_t *bson = JsonObject_to_bson(selector);
+		bson_error_t error;
+		if(!mongoc_collection_remove(recv, MONGOC_REMOVE_SINGLE_REMOVE, bson, NULL, &error)) {
+			MongoDb_set_mongoc_error(recv, &error);
+			return false;
+		}
+		return true;
+	`}
+
+	# Removes all the document that match `selector`.
+	fun remove_all(selector: JsonObject): Bool
+		import JsonObject.to_bson, set_mongoc_error `{
+		bson_t *bson = JsonObject_to_bson(selector);
+		bson_error_t error;
+		if(!mongoc_collection_remove(recv, MONGOC_REMOVE_NONE, bson, NULL, &error)) {
+			MongoDb_set_mongoc_error(recv, &error);
+			return false;
+		}
+		return true;
+	`}
+
+	# Updates a document already existing in the collection.
+	#
+	# No upsert is done, see `save` instead.
+	#
+	# ~~~
+	# var client = new MongoClient("mongodb://localhost:27017/")
+	# var col = client.database("test").collection("test")
+	# var sel = new JsonObject
+	# sel["foo"] = 10
+	# var upd = new JsonObject
+	# upd["bar"] = "BAR"
+	# assert col.update(sel, upd)
+	# client.close
+	# ~~~
+	fun update(selector: JsonObject, update: JsonObject): Bool
+		import JsonObject.to_bson, set_mongoc_error `{
+		bson_t *sel = JsonObject_to_bson(selector);
+		bson_t *upd = JsonObject_to_bson(update);
+		bson_error_t error;
+		if(!mongoc_collection_update(recv, MONGOC_UPDATE_NONE, sel, upd, NULL, &error)) {
+			MongoDb_set_mongoc_error(recv, &error);
+			return false;
+		}
+		return true;
+	`}
+
+	# Updates all documents matching the `selector`.
+	fun update_all(selector: JsonObject, update: JsonObject): Bool
+		import JsonObject.to_bson, set_mongoc_error `{
+		bson_t *sel = JsonObject_to_bson(selector);
+		bson_t *upd = JsonObject_to_bson(update);
+		bson_error_t error;
+		if(!mongoc_collection_update(recv, MONGOC_UPDATE_MULTI_UPDATE, sel, upd, NULL, &error)) {
+			MongoDb_set_mongoc_error(recv, &error);
+			return false;
+		}
+		return true;
+	`}
+
+	# Counts the document matching `query`.
+	#
+	# Returns `-1` if an error occured. See `Sys::last_mongoc_error`.
+	#
+	# ~~~
+	# var client = new MongoClient("mongodb://localhost:27017/")
+	# var col = client.database("test").collection("test")
+	# var query = new JsonObject
+	# query["foo"] = 10
+	# assert col.count(query) > 0
+	# client.close
+	# ~~~
+	fun count(query: JsonObject): Int
+		import JsonObject.to_bson, set_mongoc_error `{
+		bson_t *q = JsonObject_to_bson(query);
+		bson_error_t error;
+		int64_t count = mongoc_collection_count(recv, MONGOC_QUERY_NONE, q, 0, 0, NULL, &error);
```

Comment:
There might be a transitive import of `<stdint.h>` but just in case, if the library evolves at some point and removes the import, you might as well add it in the import section (and I like the idea that it'll give you a bit more work :) )